guides/web-apps/acceptance-testing-with-selenium/README.md

```diff
@@ -114,157 +84,61 @@ We don't want to test only Firefox. Because of that, you should use Selenium Rem
+When you rebuild the test project now, you should see each test for every browser.
 
+The final setup looks like this:
 
-```cs
-private string browser;
+![visual studio selenium screenshot](/assets/2016-04-01-13_51_26-Launcher-Microsoft-Visual-Studio.png)
 
-public PageLoads(string browser)
-{
-    this.browser = browser;
-}
-```
+Before you can execute the tests, start Selenium Server Standalone by calling `java -jar selenium-server-standalone-3.*.jar`.
 
-When you rebuild the test project now, you should see 3 tests in the Test Explorer, each test for every browser.
+## Wait for asynchronous content
 
-The final setup looks like this:
+There is one common pitfall when writing Selenium tests. The test is executed with disregard of the asynchronously loaded content. This means that your tests need to explicitly wait for UI elements to appear before running any actions or assertions on them.
 
-![visual studio selenium screenshot](/assets/2016-04-01-13_51_26-Launcher-Microsoft-Visual-Studio.png)
+It is a good practice to always wait:
 
-Before you can execute the tests, start Selenium Server Standalone by calling `java -jar selenium-server-standalone-3.0.0-beta3.jar`.
+- Wait for a text element to be present before you check the content of that element
 
-Now, press "Run All" and relax. This will take a while as the system runs your test in three different browsers!
+	- An **example** can be found [here](https://github.com/StarcounterApps/KitchenSink/blob/master/test/KitchenSink.Tests/Test/SectionString/TextareaPageTest.cs)	
+	- Please take a look at [TextareaPage_WriteToTextArea()](https://github.com/StarcounterApps/KitchenSink/blob/master/test/KitchenSink.Tests/Test/SectionString/TextareaPageTest.cs#L29) test and [line 37](https://github.com/StarcounterApps/KitchenSink/blob/master/test/KitchenSink.Tests/Test/SectionString/TextareaPageTest.cs#L37).	
+	- Method `WaitForText()` is use to wait for text in `TextareaInfoLabel` for maximum 5 second. Then assert is true and test pass.
 
-![test explorer screenshot](/assets/2016-04-01-15_38_44-Launcher-Microsoft-Visual-Studio.png)
+- Wait for a button to be present before you click on that button
 
-## Wait for asynchronous content
+	- An **example** can be found [here](https://github.com/StarcounterApps/KitchenSink/blob/master/test/KitchenSink.Tests/Test/SectionNumber/ButtonPageTest.cs)	
+	- As you can see in [line 35](https://github.com/StarcounterApps/KitchenSink/blob/master/test/KitchenSink.Tests/Test/SectionNumber/ButtonPageTest.cs#L35) in [ButtonPage_RegularButton()](https://github.com/StarcounterApps/KitchenSink/blob/master/test/KitchenSink.Tests/Test/SectionNumber/ButtonPageTest.cs) test we use one of the helper method called `WaitUntil`.	
+	- As parameter we use `Displayed` property for the button. The test will wait until `ButtonInlineScript` is displayed. Default maximum wait time is 10 second.
```

Comment:
- An **example** can be found in the method `ButtonPage_RegularButton` in the KitchenSink tests (see [ButtonPageTest.cs lines 29-46](https://github.com/StarcounterApps/KitchenSink/blob/master/test/KitchenSink.Tests/Test/SectionNumber/ButtonPageTest.cs#L29-L46)). The method `WaitUntil()` is used to asynchronously check the state of the `Displayed` property of a button. It halts the test for a default maximum time of 10 seconds. If the button is not displayed within that time, it throws an exception.